ann_visualizer/visualize.py

```diff
@@ -27,180 +27,186 @@ def ann_viz(model, view=True, filename="network.gv", title="My Neural Network"):
-                elif (type(layer) == keras.layers.core.Activation):
-                    layer_types.append("Activation");
+                hidden_layers.append(1)
+                if type(layer) == Conv2D:
+                    layer_types.append("Conv2D")
+                elif type(layer) == MaxPooling2D:
+                    layer_types.append("MaxPooling2D")
+                elif type(layer) == Dropout:
+                    layer_types.append("Dropout")
+                elif type(layer) == Flatten:
+                    layer_types.append("Flatten")
+                elif type(layer) == Activation:
+                    layer_types.append("Activation")
         else:
-            if(layer == model.layers[-1]):
-                output_layer = int(str(layer.output_shape).split(",")[1][1:-1]);
+            if (layer == model.layers[-1]):
+                output_layer = int(str(layer.output_shape).split(",")[1][1:-1])
             else:
-                hidden_layers_nr += 1;
-                if (type(layer) == keras.layers.core.Dense):
-                    hidden_layers.append(int(str(layer.output_shape).split(",")[1][1:-1]));
-                    layer_types.append("Dense");
+                hidden_layers_nr += 1
+                if (type(layer) == Dense):
+                    hidden_layers.append(int(str(layer.output_shape).split(",")[1][1:-1]))
+                    layer_types.append("Dense")
                 else:
-                    hidden_layers.append(1);
-                    if (type(layer) == keras.layers.convolutional.Conv2D):
-                        layer_types.append("Conv2D");
-                    elif (type(layer) == keras.layers.pooling.MaxPooling2D):
-                        layer_types.append("MaxPooling2D");
-                    elif (type(layer) == keras.layers.core.Dropout):
-                        layer_types.append("Dropout");
-                    elif (type(layer) == keras.layers.core.Flatten):
-                        layer_types.append("Flatten");
-                    elif (type(layer) == keras.layers.core.Activation):
-                        layer_types.append("Activation");
-        last_layer_nodes = input_layer;
-        nodes_up = input_layer;
-        if(type(model.layers[0]) != keras.layers.core.Dense):
-            last_layer_nodes = 1;
-            nodes_up = 1;
-            input_layer = 1;
+                    hidden_layers.append(1)
+                    if (type(layer) == Conv2D):
+                        layer_types.append("Conv2D")
+                    elif (type(layer) == MaxPooling2D):
+                        layer_types.append("MaxPooling2D")
+                    elif (type(layer) == Dropout):
+                        layer_types.append("Dropout")
+                    elif (type(layer) == Flatten):
+                        layer_types.append("Flatten")
+                    elif (type(layer) == Activation):
+                        layer_types.append("Activation")
```

Comment:
See: https://stackoverflow.com/questions/1549801/what-are-the-differences-between-type-and-isinstance?noredirect=1&lq=1